utils/requirements.ts

```diff
@@ -0,0 +1,375 @@
+import { decodeAbiParameters, encodeAbiParameters, zeroAddress } from 'viem';
+
+import {
+  Character,
+  CraftRequirement,
+  Game,
+  RequirementNode,
+} from '@/utils/types';
+
+const EMPTY_CLAIM_REQUIREMENTS_BYTES =
+  '0x0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c00000000000000000000000000000000000000000000000000000000000000000' as `0x${string}`;
+
+const EMPTY_CRAFT_REQUIREMENTS_BYTES =
+  '0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000' as `0x${string}`;
+
+export const validateNode = (
+  node: RequirementNode | null,
+  game: Game | null | undefined,
+): boolean => {
+  if (!node || !game) return false;
+
+  if (node.operator === 'NIL') {
+    if (!node.asset) return false;
+    if (
+      !node.asset.assetAddress ||
+      node.asset.assetAddress === '0x0' ||
+      node.asset.assetAddress === zeroAddress
+    )
+      return false;
+    if (!node.asset.assetId) return false;
+    if (!node.asset.amount || node.asset.amount === '0') return false;
+    if (node.asset.assetCategory === 'ERC1155') {
```

Comment:
It seems like the validation is different from `checkClaimRequirements`. In `checkClaimRequirements` it's considering the class category 'ERC721' even though it's actually an 'ERC115' as well

Reply:
fixed in [846ae47](https://github.com/raid-guild/character-sheets-frontend/pull/170/commits/846ae471e0454be0c7b863960a53c5166866b938)